core/test/base/iterator_factory.cpp

```diff
@@ -511,4 +512,141 @@ TYPED_TEST(PermuteIterator, DecreasingIterator)
 }
 
 
+template <typename ValueType>
+class TransformIterator : public ::testing::Test {
+protected:
+    using value_type = ValueType;
+};
+
+TYPED_TEST_SUITE(TransformIterator, gko::test::ComplexAndPODTypes,
+                 TypenameNameGenerator);
+
+
+TYPED_TEST(TransformIterator, EmptyIterator)
+{
+    auto test_iter = gko::detail::make_transform_iterator<TypeParam*>(
+        nullptr, [](TypeParam v) { return v; });
+
+    ASSERT_NO_THROW((void)std::find(test_iter, test_iter, TypeParam{}));
```

Comment:
does it check anything? `std::find` does not do anything because first and end are the same.
also what is `(void)` for?

Reply:
No, this is just a copy of the same test for permute_iterator. I needed to use a read-only operation, so find was the first thing that came to mind. The (void) is there to silence the `[[nodiscard]]` warning.